src/datamodel/serializer/Uuid.cpp

```diff
@@ -10,3 +10,65 @@
 // SPDX-License-Identifier: Apache-2.0
 
 #include "up-cpp/datamodel/serializer/Uuid.h"
+
+#include <iomanip>
+#include <sstream>
+#include <stdexcept>
+
+namespace uprotocol::datamodel::serializer::uuid {
+
+std::string AsString::serialize(const uprotocol::v1::UUID uuid) {
+	std::stringstream ss;
+	ss << std::hex << std::setw(16) << std::setfill('0') << uuid.msb() << "-"
+	   << std::setw(16) << std::setfill('0') << uuid.lsb();
+	return ss.str();
```

Comment:
Changing this to `return std::move(ss).str();` tells the compiler that `ss` is no longer needed, allowing the returned string to be move constructed. This saves an allocate, copy, deallocate cycle (so long as the string is longer than the _small string optimization_ length of a given build environment (around 12-26 characters, depending on various factors).

```suggestion
	return std::move(ss).str();
```

Note that this is only needed here because the function returns the output of a method on `ss`. Normally the compiler can figure this out itself if directly returning a local variable (via _return value optimization_).